leaf-ui/js/loadSaveFunctions.js

```diff
@@ -37,41 +36,24 @@ reader.onload = function() {
  * @param {Object} obj
  */
 function loadFromObject(obj) {
-	model = new Model();
-	model.actors = getActorsArr(obj.model.actors);
-	model.intentions = getIntentionsArr(obj.model.intentions);
-	model.links = getLinksArr(obj.model.links);
-	model.constraints = getConstArr(obj.model.constraints);
-	model.maxAbsTime = obj.model.maxAbsTime;
-
-	// store deep copy of model for detecting model changes
-	// copy is NOT of type Model
-    var previousModel = JSON.parse(JSON.stringify(model));
-
-	// Clear any previous analysis data 
-	if (analysisMap.size != 0) {
-		analysisMap.clear();
-		currAnalysisConfig = null;
-	}
-
-	// If the object contains analysis, create analysis fields from JSON
-	if (obj.analysisMap != undefined) {
-		// Parse analysis as map
-		var tempMap = new Map(Object.entries(obj.analysisMap));
-		// Loop through all analysis configs
-		for(let configObj of tempMap.values()) {
-			var config = new AnalysisConfiguration(configObj.id, new AnalysisRequest(configObj.analysisRequest), configObj.initialPosition);
-			config.setResults(configObj.analysisResults);
-			// Add config to the global analysisMap
-			analysisMap.set(config.id, config);
+	graph.fromJSON(obj.graph);
+	var cells = graph.getCells();
+	for (var i = 0; i < cells.length; i++) {
+		cell = cells[i];
+		if (cell.get('type') == "basic.Actor"){
+			createBBActor(cell) //Create actor
+		}else if (cell.get('type') == "basic.CellLink") {
+			createBBLink(cell) //Create link
+		}else{
+			// Singled out functionSegList from obj as it doesn't show up in the graph after reading from JSON
+			var funcseg = obj.graph.cells[i].intention.attributes.evolvingFunction.attributes.functionSegList;
+			createBBElement(cell, funcseg) //Create element
 		}
-	} else {
-		// Else if no analysisMap param, grab the analysisRequest
-		analysisRequest = Object.assign(new AnalysisRequest, obj.analysisRequest);
-		
 	}
-
-	graph.fromJSON(obj.graph);
+	// If the object contains configCollection, create configCollection fields from JSON
+	if (obj.configCollection != undefined) {
```

Comment:
What does the if statement exactly do?

Reply:
Try to determine if it needs to read the configCollection in